scripts/linux/smokeTestHelper.sh

```diff
@@ -0,0 +1,220 @@
+  #
+  # $1 - wait time in seconds
+
+  waitAttempts=$1
+  sleepDuration=10
+  for (( i=0; $i<$waitAttempts; i++ ))
+  do
+    if [[ $(ps aux | grep -c apt.systemd.daily) == 1 ]]; then
+      break
+    fi
+    sleep $sleepDuration
+  done
+}
+
+setup-focal-source-apt()
+{
+  # 1ES image for Ubuntu20.04 (Focal) actually has a misconfigured apt repo source
+  # This function correct the repo source by directly configure the apt to point to 
+  # the Microsoft Linux Repository for Focal.
+
+  for file in $(ls /etc/apt/sources.list.d/*.list)
+  do
+    sudo sed -i 's/bionic/focal/g' $file
+  done
+
+  sudo sed -i 's/bionic/focal/g' /etc/apt/sources.list && \
+  sudo rm -f /etc/apt/sources.list.d/microsoft-prod.list && \
+  wget https://packages.microsoft.com/config/ubuntu/20.04/prod.list  -O /etc/apt/sources.list.d/microsoft-prod.list && \
+  sudo apt update -y && \
+  sudo DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" dist-upgrade && \
+  sudo apt update -y && \
+  sudo apt upgrade -y
+}
+
+setup-config-apt()
+{
+  # Download the `apt` config file from the provided uri ($1), install it, and update the apt
+  #
+  # $1 - uri for the config deb package
+  # Ref: https://docs.microsoft.com/en-us/azure/iot-edge/how-to-provision-single-device-linux-symmetric?view=iotedge-2020-11&tabs=azure-portal%2Cubuntu
+
+  # For ARM machine, we need to wait for the apt-get daily update to be done first before we can proceed.
+  wait-for-dpkg-lock 120
+
+  echo "Setup source artifact repository:"
+  echo "$1"
+  wget "$1" -O packages-microsoft-prod.deb &&
+  sudo dpkg --force-confnew -i packages-microsoft-prod.deb &&
+  rm packages-microsoft-prod.deb &&
+  sudo apt-get update -y
+}
+
+test-released-metadata()
+{
+  # Amalgam of test cases to test PMC or GitHub release page.
+  #
+  # $1 - package name (string)
+  # $2 - version string (string)
+
+  pkgName=$1
+  version=$2
+
+  echo "Get the latest version string from linux repository"
+  # Note: $latestVersionApt includes the trailing revision: "1.2.10-1"
+  latestVersionApt=$(get-latest-version-apt "$pkgName" 1)
+  check-matching-version "$latestVersionApt" "$version" "$pkgName"
+}
+
+test-released-artifact()
+{
+  # Amalgam of test cases to test PMC or GitHub release page.
+  #
+  # $1 - package name (string)
+  # $2 - artifact version string (string)
+
+  # Trying my best to make sure the script can easily be updated to run locally
+  pkgName=$1
+  artifactVersion=$2
+  artifactPath=$3
+  gitHubArtifactSuffix=$4
+  sizePercentThrehold=$5
+  isCheckPreviousPkg=$6
+
+  echo "Download PMC artifacts ($pkgName)"
+  wgetRespCurrent=$(download-artifact-from-pmc-apt "$pkgName" "$artifactPath")
+  echo $wgetRespCurrent
+  currentArtifact=${wgetRespCurrent##* }
+  echo "[PASS] The artifacts can be downloaded from the Microsoft Linux Package Repository"
+  echo "The releasing artifact: $currentArtifact"
+  echo ""
+
+  if $isCheckPreviousPkg; then
```

Comment:
What is motivation for being optional?

Reply:
In case, release 1.3.0, it's the first release without any prior pkg to be compared.